### DISCUSSION TITLE
survminer包使用的疑问

### DISCUSSION THREAD

**Discussion by zhaoliang0302** [ShixiangWang/self-study#9] [Q&A] [answered]
1.我看了<http://r-addict.com/2016/11/21/Optimal-Cutpoint-maxstat.html>这个文章后尝试给我的表达矩阵做生存分析cutoff，res.cut <- surv_cutpoint(svdata, time = "OS.time", event = "OS", variables = names(svdata)[1:(ncol(svdata)-2)]) ，运行了34%后，报错为：Error in cmaxstat(scores, x, weights = weights, pmethod, minprop, maxprop,  : no data between minprop, maxprop。我猜想是不是数据的某一列有很多是0，造成cutoff无法设置出来
2. res.cat <- surv_categorize(res.cut)这一步按道理来说会把基因的一列分成high和low，但是我发现凡是genename中有“-”符号的都还是表达量，我尝试把“-”换成“_”后就可以了，怎么才能不这么麻烦呢
![image](https://user-images.githubusercontent.com/42333702/50981093-2897d580-1535-11e9-9c20-cf745e101e8f.png)

**Comment by ShixiangWang** [comment]
@zhaoliang0302 

1. 你的想法是正确的，有可能存在不能划分的情况，你需要检查下是不是有这样的一些基因。
2. 最简单的办法是使用`make.names()`，这个问题应该是你的列名是非语义名造成的，所以可以用

```
colnames(res.cut) = make.names(colnames(res.cut))
```

这样的方法重命名。

**Comment by zhaoliang0302** [comment]
谢谢翔哥，
1）我用了svdata <- data[,sapply(data, function(x)
  ifelse(sum(x != 0, na.rm = TRUE) > 30, TRUE, FALSE)) ]  # 选出一列中不为0的数超过30个的数据，之后就不报错了，应该是数据很多0造成的
2）colnames(res.cut) = make.names(colnames(res.cut))我用了之后genename中的“-”会变成“."，我先用了colnames(svdata) <- str_replace_all(colnames(svdata), c("-" = "_"))后再转成”-“，问题解决了
感谢！

**Comment by ShixiangWang** [comment]
Okay, 那我就关闭issue了。后续如果你还有问题就打开它补充说明。

**Comment by zoucheng123** [comment]
在surv_cutpoint这一步，遇到报错“Error in maxstat(y = c(71.8, 16, 1.9, 4, 34.6, 56, 67.5, 41.1, 74.3, 36.8,  :    pmethod not specified.”   请问是什么原因呢？
十分感谢！

**Comment by ShixiangWang** [comment, marked as answer]
@zoucheng123 你没有指定计算 p 值的方法，请查看函数文档及其示例。
